utils/managed_bytes.hh

```diff
@@ -546,7 +546,7 @@ inline managed_bytes::managed_bytes(const managed_bytes& o) {
     if (o.is_inline()) {
         _inline_size = o._inline_size;
         _u = o._u;
-    } else if (o.is_single_chunk()) {
+    } else if (o.is_single_chunk() && o.size() <= max_seg(current_allocator())) {
```

Comment:
@michoecho Is there an off-by-one here? managed_bytes(initialized_later, size_type size) thinks o.size() == max_seg is multi chunked.  Maybe the off-by-one is there, since size == max_seg should be fine AFAIU.

Reply:
Good eye. Yes, there's an "off-by-one" in `managed_bytes(initialized_later, size_type size)`. If `size == max_seg`, then a the single-chunk layout could be used, but isn't.

I knew about it, but decided not to touch it, because it doesn't matter.

Reply:
In fact, it's not "off-by-1", but "off-by-17", because `max_seg` is the "pessimistic" size per chunk, which assumes that the multi-chunk metadata is used. But the single-chunk metadata is smaller by 16 bytes, so you can fit `max_seg + 16` bytes into the allocation.